engine/access/rpc/handler.go

```diff
@@ -894,15 +895,27 @@ func (h *Handler) GetEventsForHeightRange(
 
 	eventEncodingVersion := req.GetEventEncodingVersion()
 
-	results, err := h.api.GetEventsForHeightRange(ctx, eventType, startHeight, endHeight, eventEncodingVersion)
+	results, executorMetadata, err := h.api.GetEventsForHeightRange(
+		ctx,
+		eventType,
+		startHeight,
+		endHeight,
+		eventEncodingVersion,
+		optimistic_sync.NewCriteria(req.GetExecutionStateQuery()),
```

Comment:
how about creating an `ExecutionStateQuery` object under `models/access/`, then adding a converter for it. That way you could use the same object in the REST API, and update `NewCriteria` to use the common object.